### ISSUE TITLE
ESM imports of @babel/runtime in CJS builds

### ISSUE THREAD

**Issue by StringEpsilon** [wessberg/rollup-plugin-ts#15]
When using the babel transpiler, CJS builds are broken because the plugin forces babel to use the ESM modules. Meaning you get imports like these in the output:

```js
var _objectWithoutPropertiesLoose = _interopDefault(require('@babel/runtime/helpers/esm/objectWithoutPropertiesLoose')), 
_objectSpread = _interopDefault(require('@babel/runtime/helpers/esm/objectSpread')), 
_inheritsLoose = _interopDefault(require('@babel/runtime/helpers/esm/inheritsLoose')), 
```

This potentially breaks consumers of the CJS module in question. In my case, I want to run my unit tests against the CJS module and jest throws an exception.

The unfortunate behavior is cause here:

https://github.com/wessberg/rollup-plugin-ts/blob/2c8ea0c685e25f6d11b1bca3c74ffbe1b0889a79/src/constant/constant.ts#L69-L73

useESModules should only be set to true, if the rollup output format is ESM.

**Comment by wessberg** [comment]
Hi there, thanks for reporting this issue. The reason for forcing ESM is to delegate all responsibility of module emission to Rollup, which may or may not emit cjs, ESM, or any other supported module format. Rollup has first-class support for ES modules, and if cjs Babel helpers are being used, Rollup will need to be used in conjunction with `rollup-plugin-commonjs` which can be avoided by using ESM helpers. The issue you are reporting here is very valid indeed, and I will look into it. What's strange here is that the `require` calls are preserved, but they shouldn't - instead, the referenced helpers should be inlined in the bundle.

Can you share your Rollup and Babel configuration with me? I'll look into it as soon as possible. Thanks!

**Comment by StringEpsilon** [comment]
My rollup config:

**Comment by StringEpsilon** [comment]
Thank you very much for the swift work!

**Comment by wessberg** [comment]
Alright, I've reproduced the issue based on your project and configs.
The reason for this issue is that the babel helpers are marked as external by your Rollup config. `isBareModuleId` returns true for all babel helpers such as `@babel/runtime/helpers/esm/typeof.js`. This means that the helpers won't be inlined in the bundle. The current setup where `useESModules` are forced is based on the assumption that all helpers will be inlined.

To fix it, I've added a check that only forces the option if those helpers are going to be inlined (e.g. if they are not marked as external). If not, it will respect your babel config and use whatever option you gave.

I've released it as v1.1.46 👍
